tests/core/dom/range/clonecontents.js

```diff
@@ -22,6 +22,11 @@
 			var range,
 				clone;
 
+			// Prevent selection optimizations from breaking tests (#3175).
```

Comment:
These test cases are build around selection which might be not possible after selection optimizations. Optimization would apply to initial selection. These tests are to check if other things than optimizations have effect on selection, so turning off optimizations is required to test for regression caused by other methods.

For instance, commented code relates to a [bug](https://dev.ckeditor.com/ticket/11586) when `range.cloneContents` altered selection. In this case turning off optimizations for TC is the way check for regression without rewriting test. Also note that optimization are applied to editor selection, not just a range, so you can create a range without selecting it, and call `range.cloneContents()`.

IMO turning off optimization in these tests is easiest and safest way to keep these tests. Otherwise we should go back to each TC and analyse it, analyse initial issue, and rewrite it in a way that optimizations doesn't affect it.

>The thing I'm concerned about is that if preventing optimizations here will not mask any issues (unit test will pass while in reality it will fail)
This won't be the same case, if the test passes without optimization, but something is wrong with it, then the optimization itself could be source of the new issue, hence it's different case.

My last argument for keeping these tests in current shape is that in future we might add an option to turn off optimizations.